### PR TITLE
[docs] Add differences section to implement API documentation

### PR DESCRIPTION
**Pull request by felixsch**

### CODE REVIEW

File: doc/SCC-API-(Implemented).md [hunk 1]
```diff
@@ -12,6 +12,73 @@ These several levels of authentication also act as scopes on the queried
...
+    * `/connect/systems/products/migrations`
+    * `/connect/systems/products/synchronize`
+    * `/connect/systems/activations`
+
+- No `registration code` is required by RMT.
+  *Affected endpoints*:
+    * `/connect/subscriptions/products`
+
+- A `registration code` is not required by SMT. (If its supplied it will show up
+  in SCC if system information forwarding is enabled in SCC)
```
Comment (lagartoflojo) [inline doc/SCC-API-(Implemented).md]: > forwarding is enabled in ~~SCC~~ SMT

File: doc/SCC-API-(Implemented).md [hunk 2]
```diff
@@ -12,6 +12,73 @@ These several levels of authentication also act as scopes on the queried
...
+
+- The `available` attribute is only present in SMT/RMT and indicates if a certain
+  repository is mirrored or not.
+  *Affected endpoints*:
+    * `/connect/*/products`
+    * `/connect/systems/services`
+    * `/connect/systems/products/synchronize`
+    * `/connect/systems/activations`
+
+- The product `free` attribute is always true in RMT.
```
Comment (lagartoflojo) [inline doc/SCC-API-(Implemented).md]: And in SMT?

File: doc/SCC-API-(Implemented).md [hunk 3]
```diff
@@ -12,6 +12,73 @@ These several levels of authentication also act as scopes on the queried
...
+- No `registration code` is required by RMT.
+  *Affected endpoints*:
+    * `/connect/subscriptions/products`
+
+- A `registration code` is not required by SMT. (If its supplied it will show up
+  in SCC if system information forwarding is enabled in SCC)
+  *Affected endpoints*:
+    * `/connect/subscriptions/products`
+
+- The `distro_target` attribute is only used by SMT.
```
Comment (lagartoflojo) [inline doc/SCC-API-(Implemented).md]: By "used" do you mean it consumes that attribute, or it adds it to the JSON responses? Maybe you could clarify.

File: doc/SCC-API-(Implemented).md [hunk 4]
```diff
@@ -12,6 +12,73 @@ These several levels of authentication also act as scopes on the queried
...
+  products instead of a service for each product.
+  *Affected endpoints*:
+    * `/connect/systems/activations`
+
+- Only SCC implements the package search functionality
+  *Affected endpoints:*
+    * `/api/products/search_packages`
+
+- RMT does not implement all API endpoints of the SCC API.
+  *Endpoints are available in RMT:*
```
Comment (lagartoflojo) [inline doc/SCC-API-(Implemented).md]: Aren't these "endpoints **not** available in RMT"?

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/131130/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 72e3022a0f637139929859ac96fde66501181b8f into 7f743bc184a87b995ea2f02008cdb3b22387b839)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/131138/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged e20d3c76ce59afdc11498378f5a364411475cbd8 into 7f743bc184a87b995ea2f02008cdb3b22387b839)

File: doc/SCC-API-(Implemented).md [hunk 5]
```diff
@@ -12,6 +12,74 @@ These several levels of authentication also act as scopes on the queried
...
+    * `/connect/systems/services`
+    * `/connect/systems/products/migrations`
+    * `/connect/systems/products/synchronize`
+    * `/connect/systems/activations`
+
+- No `registration code` is required by RMT.
+  *Affected endpoints*:
+    * `/connect/subscriptions/products`
+
+- A `registration code` is not required by SMT. (If its supplied it will show up
```
Comment (lagartoflojo) [inline doc/SCC-API-(Implemented).md]: One little nitpick: `it's` not `its`

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/131417/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 9132c7b2c24ac1c08fe360ff122ddbbd1fc940ec into 7f743bc184a87b995ea2f02008cdb3b22387b839)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/131445/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged f520920765ab56b53a1044708acf5e4a1dc19767 into 7f743bc184a87b995ea2f02008cdb3b22387b839)

**Pull request merged**